Automattic/simplenote-ios
# Editor: Drops direct ivar indirections

### Details:
This PR does **not introduce new behavior**:
- Replaces old school **ivars** with actual ObjC properties
- Moves the protocol conformance list to the Editor's private category
- Drops direct ivar indirections: `self->_ivar` 🔥🔥
- Drops several dead properties

I just couldn't let go another day without dropping those `->_`!!

@aerych Sir!! may I bug you a biiiiit?
**Thank you!!!**


### Test
**Please** smoke test the editor:

- [x] Verify the Navigation Bar looks as expected
- [x] Verify you can swipe to dismiss
- [x] Verify you can swipe (from the right edge) to push Markdown Preview (on a Markdown-Enabled note)
- [x] Verify the top right New Note button adds a new document
- [x] Verify the top right `( i )` icon opens the Info Sheet
- [x] Verify that once the keyboard is onscreen, you can dismiss it with the navigation bar's Keyboard icon

### Test: Search
1. Open the Notes List
2. Press over the searchbar and type a keyword that yields results
3. Press over any of the Notes in your list

- [x] Verify that you can navigate across search results with the Search Bar located at the bottom
- [x] Verify the back button reads as `< Search`
- [x] Verify that pressing `Done` dismisses the Search Mode 

### Release
These changes do not require release notes.
